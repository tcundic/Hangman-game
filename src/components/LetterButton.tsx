import React, {MouseEventHandler} from "react";

const LetterButton = ({letter, used, onClick}: { letter: string, used: boolean, onClick: MouseEventHandler }) => {
    return (
        <div className="column is-inline-touch">
            {used ?
                <button className="button is-info is-light is-large my-4">{letter}</button> :
                <button onClick={onClick} className="button is-info is-large my-4">{letter}</button>
            }
        </div>
    );
};

export default LetterButton;