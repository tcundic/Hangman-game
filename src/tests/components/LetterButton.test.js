import React from "react";
import { shallow } from "enzyme";
import LetterButton from "../../components/LetterButton";

const onClick = () => {};

test('should render LetterButton correctly', () => {
    const wrapper = shallow(<LetterButton letter="A" used={false} onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render LetterButton correctly when letter is already used', () => {
    const wrapper = shallow(<LetterButton letter="A" used={true} onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
});
