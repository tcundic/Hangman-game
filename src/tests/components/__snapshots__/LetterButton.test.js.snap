// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LetterButton correctly 1`] = `
<div
  className="column is-inline-touch"
>
  <button
    className="button is-info is-large my-4"
    onClick={[Function]}
  >
    A
  </button>
</div>
`;

exports[`should render LetterButton correctly when letter is already used 1`] = `
<div
  className="column is-inline-touch"
>
  <button
    className="button is-info is-light is-large my-4"
  >
    A
  </button>
</div>
`;
